### ISSUE TITLE
Verify test package name when generating tests

### ISSUE THREAD

**Issue by SteelPhase** [cweill/gotests#131]
I've noticed that when generating tests, they will generate as invalid if the `*_test.go` file uses the `_test` suffix in its package name (i.e.`package *_test`).

**Comment by cweill** [comment]
I believe the convention is to save the *_test suffix for tests only, so this is working as intended, unless you have a concrete example of when this functionality is needed.

**Comment by SteelPhase** [comment]
I've heard it to referred to as external testing, but [this page covers the functionality](https://tip.golang.org/cmd/go/#hdr-Test_packages). I'm sure there's more info available



> Test files that declare a package with the suffix "_test" will be compiled as a separate package, and then linked and run with the main test binary.


There is an issue on functionality documentation [here](https://github.com/golang/go/issues/25223)

**Comment by cweill** [comment]
I see, this is used for test-only packages right? If you want to send a PR to fix this, I'll be happy to approve.

**Comment by SteelPhase** [comment]
Yes it is, I'll see if I can come up with a simple way to solve this.
